packages/components/src/DataList/components/DataListFilters/components/DataListSort/DataListSort.tsx

```diff
@@ -53,20 +55,19 @@ export function DataListSort() {
           customOptions.forEach(option => {
             acc.push({
               label: option.label || "",
-              value: `${sort.key},${option.order},${option.label}`,
+              value: `${sort.key},${option.order},${option.label},${option.dataValue}`,
```

Comment:
maybe put this in a function so you can have 1 source of truth

Reply:
I've split out some logic where we were setting these values in multiple places - but this file needs a good clean up so I'll put more effort into that in my cleanup PR